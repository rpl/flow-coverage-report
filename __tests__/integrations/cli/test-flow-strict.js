'use babel';

import path from 'path';

import {
  FIXTURE_PATH,
  runFlowCoverageReport
} from '../../common';

const testProjectDir = path.join(FIXTURE_PATH, 'flow-strict');

jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000; // 10 second timeout

test('Accept \'@flow strict\' and \'@flow strict-local\' pragmas', async () => {
  const {stdout, stderr} = await runFlowCoverageReport([
    '-i', `"src/*.js"`
  ], {cwd: testProjectDir});

  const filteredStdoutMain = stdout.split('\n').filter(line => line.indexOf('src/main') >= 0);
  const filteredStdoutLocal = stdout.split('\n').filter(line => line.indexOf('src/local') >= 0);

  expect({filteredStdoutMain, filteredStdoutLocal, stderr}).toMatchSnapshot();
});
