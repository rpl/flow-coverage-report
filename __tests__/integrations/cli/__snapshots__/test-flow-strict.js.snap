// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accept '@flow strict' and '@flow strict-local' pragmas 1`] = `
Object {
  "filteredStdoutLocal": Array [
    "│ src/local.js │ flow strict-local │   100 % │     0 │       0 │ 0         │",
  ],
  "filteredStdoutMain": Array [
    "│ src/main.js  │       flow strict │   100 % │     0 │       0 │ 0         │",
  ],
  "stderr": "",
}
`;
